### PR TITLE
Add an option to disable comments from collections

### PR DESCRIPTION
**Pull request by ishunko**
Resolves https://nanohub.org/support/ticket/362351

### CODE REVIEW

File: core/components/com_collections/site/controllers/posts.php [hunk 1]
```diff
@@ -268,6 +268,12 @@ public function saveTask()
 	 */
 	public function savecommentTask()
 	{
+		// Error if comments are not allowed
+		if (!$this->config->get('allow_comments'))
+		{
+			App::abort(404, Lang::txt('COM_COLLECTIONS_ERROR_COMMENTS_DISALLOWED'));
```
Comment (rw2) [inline core/components/com_collections/site/controllers/posts.php:274]: Should be updated to have generic "page not found" text

File: core/components/com_collections/site/language/en-GB/en-GB.com_collections.ini [hunk 2]
```diff
@@ -160,6 +160,7 @@ COM_COLLECTIONS_ERROR_ACCESS_DENIED="You do not have permission to access this c
 COM_COLLECTIONS_ERROR_UPLOADING_INVALID_FILE="File has an invalid extension, it should be one of %s."
 COM_COLLECTIONS_ERROR_UPLOADING_FILE_TOO_BIG="File is too large. Max file upload size is %s"
 COM_COLLECTIONS_ERROR_UPLOADING_VIRUS="File failed virus scan."
+COM_COLLECTIONS_ERROR_COMMENTS_DISALLOWED="Comments are not allowed."
```
Comment (rw2) [inline core/components/com_collections/site/language/en-GB/en-GB.com_collections.ini:163]: Delete this, per Ilya

File: core/plugins/members/collections/views/post/tmpl/default.php [hunk 3]
```diff
@@ -13,6 +13,10 @@
 
 $this->css()
      ->js();
+
+// Get the comments config value
+$allow_comments = Component::params('com_collections')->get('allow_comments');
+$allow_comments;
```
Comment (rw2) [inline core/plugins/members/collections/views/post/tmpl/default.php:19]: delete this line

File: core/libraries/Hubzero/Config/Registry.php [hunk 4]
```diff
@@ -265,6 +265,95 @@ public function set($path, $value, $separator = null)
...
 	}
 
+	/**
+	 * Delete a registry value
+	 *
+	 * @param   string  $path  Registry Path
+	 * @return  mixed  The value of the removed node or null if not set
+	 *
+	 */
+	public function remove($path)
```
Comment (ishunko) [inline core/libraries/Hubzero/Config/Registry.php:275]: Remove remove method as it is not desirable.

File: core/migrations/Migration20210105000001ComCollections.php [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+			{
+				$params->set('allow_comments', 1);
+				$query = "UPDATE `#__extensions` SET `params`=" . $this->db->quote($params->toString()) . " WHERE `name`= 'com_collections' AND `element` = 'com_collections'";
+				$this->db->setQuery($query);
+				$this->db->query();
+			}
+		}
+	}
+
+	public function down()
```
Comment (ishunko) [inline core/migrations/Migration20210105000001ComCollections.php:32]: do nothing here

**Pull request merged**